File: app/src/main/java/com/cr4sh/nhlauncher/SpecialFeatures/SpecialFeaturesActivity.java
package com.cr4sh.nhlauncher.SpecialFeatures;

import android.app.ActivityOptions;
import android.content.Intent;
import android.graphics.Color;
import android.graphics.drawable.GradientDrawable;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;

import com.cr4sh.nhlauncher.MainActivity;
import com.cr4sh.nhlauncher.R;
import com.cr4sh.nhlauncher.SpecialButtonsRecycler.NHLSpecialAdapter;
import com.cr4sh.nhlauncher.SpecialButtonsRecycler.NHLSpecialItem;
import com.cr4sh.nhlauncher.utils.NHLPreferences;
import com.cr4sh.nhlauncher.utils.VibrationUtils;

import java.util.ArrayList;
import java.util.List;

public class SpecialFeaturesActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
//        overridePendingTransition(R.anim.cat_appear, R.anim.cat_disappear);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
            overrideActivityTransition(OVERRIDE_TRANSITION_OPEN, R.anim.cat_appear, R.anim.cat_appear);
        } else {
            overridePendingTransition(R.anim.cat_appear, R.anim.cat_disappear);
        }

        setContentView(R.layout.special_fragment_layout);

        NHLPreferences NHLPreferences = new NHLPreferences(this);

        RecyclerView specialRecyclerView = findViewById(R.id.special_recycler_view);

        TextView title = findViewById(R.id.textView);
        title.setTextColor(Color.parseColor(NHLPreferences.color80()));

        Button cancelButton = findViewById(R.id.cancel_button);
        cancelButton.setBackgroundColor(Color.parseColor(NHLPreferences.color80()));
        cancelButton.setTextColor(Color.parseColor(NHLPreferences.color50()));

        View rootView = findViewById(android.R.id.content);
        rootView.setBackgroundColor(Color.parseColor(NHLPreferences.color20()));
        Window window = this.getWindow();
        window.setStatusBarColor(Color.parseColor(NHLPreferences.color20()));
        window.setNavigationBarColor(Color.parseColor(NHLPreferences.color20()));

        GradientDrawable drawable = new GradientDrawable();
        if (NHLPreferences.isNewButtonStyleActive()) {
            drawable.setColor(Color.parseColor(NHLPreferences.color50()));
            drawable.setCornerRadius(60);
        } else {
            drawable.setCornerRadius(60);
            drawable.setStroke(8, Color.parseColor(NHLPreferences.color80()));
        }

        NHLSpecialAdapter nhlSpecialAdapter = new NHLSpecialAdapter();
        NHLSpecialItem specialItem = new NHLSpecialItem(getResources().getString(R.string.wps_attack), getResources().getString(R.string.wps_desc_short), "kali_fern_wifi_cracker");
        NHLSpecialItem specialItem2 = new NHLSpecialItem(getResources().getString(R.string.bt_toolkit), getResources().getString(R.string.bt_short), "kali_spooftooph");

        List<NHLSpecialItem> newItemList = new ArrayList<>();
        newItemList.add(specialItem);
        newItemList.add(specialItem2);

        nhlSpecialAdapter.updateData(newItemList);
        specialRecyclerView.setAdapter(nhlSpecialAdapter);

//        customButton1.setBackground(drawable);
////        customButton2.setBackground(drawable);
//        customButton3.setBackground(drawable);
//
//
//        customButton1.setOnClickListener(v -> {
//            VibrationUtils.vibrate(this, 10);
//            Intent intent = new Intent(this, WPSAttack.class);
//            startActivity(intent);
//        });

//        customButton2.setOnClickListener(v -> {
//            Intent intent = new Intent(this, MDKDeauth.class);
//            startActivity(intent);
//        });

//        customButton3.setOnClickListener(v -> {
//            VibrationUtils.vibrate(this, 10);
//            ToastUtils.showCustomToast(this, "Coming soon...");
////            Intent intent = new Intent(this, BluetoothAttacks.class);
////            startActivity(intent);
//        });

        cancelButton.setOnClickListener(v -> {
            VibrationUtils.vibrate(this, 10);

            Intent intent = new Intent(SpecialFeaturesActivity.this, MainActivity.class);
            intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
            Bundle animationBundle = ActivityOptions.makeCustomAnimation(
                    this,
                    R.anim.cat_appear,  // Enter animation
                    R.anim.cat_disappear  // Exit animation
            ).toBundle();
            startActivity(intent, animationBundle);
            finish();
        });

    }

    @Override
    protected void onPause() {
        super.onPause();
        if (isFinishing()) {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
                overrideActivityTransition(OVERRIDE_TRANSITION_CLOSE, R.anim.cat_appear, R.anim.cat_appear);
            } else {
                overridePendingTransition(R.anim.cat_appear, R.anim.cat_disappear);
            }
        }
    }
}

File: app/src/main/java/com/cr4sh/nhlauncher/activities/specialFeatures/SpecialFeaturesActivity.kt
package com.cr4sh.nhlauncher.activities.specialFeatures

import android.content.Intent
import android.graphics.Color
import android.graphics.drawable.GradientDrawable
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.cr4sh.nhlauncher.R
import com.cr4sh.nhlauncher.activities.MainActivity
import com.cr4sh.nhlauncher.recyclers.categoriesRecycler.specialButtonsRecycler.NHLSpecialItem
import com.cr4sh.nhlauncher.recyclers.specialButtonsRecycler.NHLSpecialAdapter
import com.cr4sh.nhlauncher.utils.ColorChanger.Companion.activityAnimation
import com.cr4sh.nhlauncher.utils.LanguageChanger
import com.cr4sh.nhlauncher.utils.NHLPreferences
import com.cr4sh.nhlauncher.utils.VibrationUtils

class SpecialFeaturesActivity : LanguageChanger() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        activityAnimation()

        setContentView(R.layout.special_fragment_layout)
        val nhlPreferences = NHLPreferences(this)
        val specialRecyclerView = findViewById<RecyclerView>(R.id.special_recycler_view)
        val title = findViewById<TextView>(R.id.textView)
        title.setTextColor(Color.parseColor(nhlPreferences.color80()))
        val cancelButton = findViewById<Button>(R.id.cancel_button)
        cancelButton.setBackgroundColor(Color.parseColor(nhlPreferences.color80()))
        cancelButton.setTextColor(Color.parseColor(nhlPreferences.color50()))
        val rootView = findViewById<View>(android.R.id.content)
        rootView.setBackgroundColor(Color.parseColor(nhlPreferences.color20()))
        val window = this.window
        window.statusBarColor = Color.parseColor(nhlPreferences.color20())
        window.navigationBarColor = Color.parseColor(nhlPreferences.color20())
        val drawable = GradientDrawable()
        if (nhlPreferences.isNewButtonStyleActive) {
            drawable.setColor(Color.parseColor(nhlPreferences.color50()))
            drawable.cornerRadius = 60f
        } else {
            drawable.cornerRadius = 60f
            drawable.setStroke(8, Color.parseColor(nhlPreferences.color80()))
        }
        val nhlSpecialAdapter = NHLSpecialAdapter()
        val specialItem = NHLSpecialItem(
            resources.getString(R.string.wps_attack),
            resources.getString(R.string.wps_desc_short),
            "wireless"
        )
        val specialItem2 = NHLSpecialItem(
            resources.getString(R.string.bt_toolkit),
            resources.getString(R.string.bt_short),
            "bluetooth"
        )
        val specialItem3 = NHLSpecialItem(
            "Net Scanner",
            resources.getString(R.string.netscanner2),
            "socat"
        )
        val newItemList: MutableList<NHLSpecialItem> = ArrayList()
        newItemList.add(specialItem)
        newItemList.add(specialItem2)
        newItemList.add(specialItem3)
        nhlSpecialAdapter.updateData(newItemList)
        specialRecyclerView.adapter = nhlSpecialAdapter
        cancelButton.setOnClickListener {
            VibrationUtils.vibrate()
            val intent = Intent(this@SpecialFeaturesActivity, MainActivity::class.java)
            intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP)
            startActivity(intent)
            finish()
        }
    }

    override fun onPause() {
        super.onPause()
        if (isFinishing) {
            activityAnimation()
        }
    }
}
